File: 02/exo2c.py
# _*_ encoding: utf-8 _*_
import sys
import json
import urllib

class NasaExo():

    def __init__(self, params):
        self.year = params[1]
        self.api_base = 'http://exoapi.com/api/skyhook/'

    def what_year(self):
        print "L'année demandée est", self.year

    def get_planets(self):
        url = self.api_base + 'planets/search?disc_year=' + self.year
        cnx = urllib.urlopen(url)
        content = cnx.read()
        decoded_planets = json.loads(content)
        return decoded_planets['response']['results']

    def get_names(self, planet_list):
        names = []
        for planet in planet_list:
            names.append(planet['name'])
        return names

    def print_list(self, my_list):
        for obj in my_list:
            print obj

exo = NasaExo(sys.argv)
planet_list = exo.get_planets()
names = exo.get_names(planet_list)
exo.print_list(names)

File: 03/exo2c.rb
#!/usr/bin/env ruby
# encoding: utf-8
require 'json'
require 'rest_client'

class NasaExo

  def initialize(params)
    @year = params[0]
    @api_base = 'http://exoapi.com/api/skyhook/'
  end

  def what_year
    puts "L'année demandée est: " + @year
  end

  def get_planets
    url = @api_base + 'planets/search?disc_year=' + @year
    content = download(url)
    decoded_planets = JSON.load(content)
    return decoded_planets['response']['results']
  end

  def get_names(planet_list)
    names = []
    planet_list.each do |planet|
      names << planet['name']
    end
    return names
  end

  def print_list(my_list)
    my_list.each do |obj|
      puts obj
    end
  end

  def download(url)
    RestClient.get(url) do |response, request, result|
      return response
    end
  end

end


exo = NasaExo.new(ARGV)

planet_list = exo.get_planets

names = exo.get_names(planet_list)

exo.print_list(names)
